mttl/models/containers/__init__.py

```diff
@@ -300,38 +313,70 @@ def add_expert_to_transformer(
 
     for m_name, module in get_modules_to_modify_trie(transformer):
         if re.fullmatch(expert_config.modify_modules, m_name):
-            for c_name, layer in dict(module.named_children()).items():
-                if re.fullmatch(expert_config.modify_layers, c_name):
-                    total_layers += 1
-                    layer_name = f"{m_name}.{c_name}"
-
-                    if not isinstance(layer, ExpertContainer):
-                        CONTAINER_CLASS = get_container_class(model_modifier)
-                        expert_container = CONTAINER_CLASS(
-                            expert_config,
-                            layer,
-                            lora_merge_after=(
-                                selector_config.lora_merge_after
-                                if selector_config
-                                else False
-                            ),
-                        )
-                        expert_container.__layer_name__ = layer_name
-                        setattr(
-                            module,
-                            c_name,
-                            expert_container,
-                        )
-                        added_containers.append(expert_container)
-                    else:
-                        expert_container = layer
-
-                    added_layers.append(expert_container.__layer_name__)
-                    expert_container.add_expert(
-                        expert,
-                        action=action,
-                        is_default=is_default,
+            if expert_config.modify_layers == "":
```

Comment:
just a q: do we really need to keep both modify_layers *and* modify_module here? isn't there something we can do to say:

modify_layers=X, 

where X is all the layers/modules that match but not their children,

e.g. if X=k_proj, we modify 1.block.k_proj, then we also modify 2.block.k_proj, cause they have different parents

but if X=block, then 1.block.k_proj, and 1.block.q_proj is modified only once (we keep track of whether we modified the parent of the current layer)

doesn't this encompass modify_layers and modify_modules?

Reply:
will remove "modify_layers" argument in a separate PR